tools/test-proxy/Azure.Sdk.Tools.TestProxy/Admin.cs

```diff
@@ -119,35 +121,49 @@ private object GenerateInstance(string typePrefix, string name, HashSet<string>
             var ctor = t.GetConstructors()[0];
             var paramsSet = ctor.GetParameters();
 
-                // walk across our constructor params. check inside the body for a resulting value for each of them
-                foreach (var param in paramsSet)
+            // walk across our constructor params. check inside the body for a resulting value for each of them
+            foreach (var param in paramsSet)
+            {
+                if (documentBody != null && documentBody.RootElement.TryGetProperty(param.Name, out var jsonElement))
                 {
-                    if (documentBody != null && documentBody.RootElement.TryGetProperty(param.Name, out var jsonElement))
+                    object argumentValue = null;
+                    switch (jsonElement.ValueKind)
                     {
-                        object valueResult = null;
-                        switch (jsonElement.ValueKind)
-                        {
-                            case JsonValueKind.Null:
-                            case JsonValueKind.String:
-                                valueResult = jsonElement.GetString();
-                                break;
-                            case JsonValueKind.True:
-                            case JsonValueKind.False:
-                                valueResult = jsonElement.GetBoolean();
-                                break;
-                            default:
-                                throw new HttpException(HttpStatusCode.BadRequest, $"{jsonElement.ValueKind} parameters are not supported");
-                        }
-
-                        if(valueResult == null || (valueResult is string stringResult && string.IsNullOrEmpty(stringResult)))
-                        {
-                            if (!acceptableEmptyArgs.Contains(param.Name))
+                        case JsonValueKind.Null:
+                        case JsonValueKind.String:
+                            argumentValue = jsonElement.GetString();
+                            break;
+                        case JsonValueKind.True:
+                        case JsonValueKind.False:
+                            argumentValue = jsonElement.GetBoolean();
+                            break;
+                        case JsonValueKind.Object:
+                            try
                             {
-                                throw new HttpException(HttpStatusCode.BadRequest, $"Parameter {param.Name} was passed with no value. Please check the request body and try again.");
+                                argumentValue = Activator.CreateInstance(param.ParameterType, new List<object> { jsonElement }.ToArray());
                             }
+                            catch (Exception e)
+                            {
+                                if (e.InnerException is HttpException)
```

Comment:
Why would there be an HttpException from Activator.CreateInstance?

Reply:
Because the constructor is designed to throw a relevant exception when it encounters a problem! 

1. Bad regex value passed in.
2. NO condition set